packages/scripts/config/webpack.config.js

```diff
@@ -50,23 +50,26 @@ const hasExperimentalModulesFlag = getAsBooleanFromENV(
 );
 
 /**
- * The plugin recomputes the render paths once on each compilation. It is necessary to avoid repeating processing
+ * The plugin recomputes PHP file paths once on each compilation. It is necessary to avoid repeating processing
  * when filtering every discovered PHP file in the source folder. This is the most performant way to ensure that
  * changes in `block.json` files are picked up in watch mode.
  */
-class RenderPathsPlugin {
+class PhpFilePathsPlugin {
 	/**
-	 * Paths with the `render` props included in `block.json` files.
+	 * PHP file paths from `render` and `variations` props found in `block.json` files.
 	 *
 	 * @type {string[]}
 	 */
-	static renderPaths;
+	static paths;
 
 	apply( compiler ) {
 		const pluginName = this.constructor.name;
 
 		compiler.hooks.thisCompilation.tap( pluginName, () => {
-			this.constructor.renderPaths = getRenderPropPaths();
+			this.constructor.paths = getPhpFilePaths( [
+				'render',
+				'variations',
```

Comment:
Maybe obtain these from constructor argument?

Reply:
No, let's follow the example of the functions in `block-json.js` and keep these hard-wired.